snapshot/src/test/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStoreTest.java

```diff
@@ -128,6 +128,66 @@ public void shouldDeleteSnapshotWithoutChecksumFile() throws IOException {
     assertThat(persistedSnapshot.getDirectory()).doesNotExist();
   }
 
+  @Test
+  public void shouldDeleteOlderSnapshotsWithCorruptChecksums() throws IOException {
+    // given
+    final var otherStore = createStore(snapshotsDir, pendingSnapshotsDir);
+    final var corruptOlderSnapshot =
+        (FileBasedSnapshot) takeTransientSnapshot(1, otherStore).persist().join();
+    SnapshotChecksum.persist(corruptOlderSnapshot.getChecksumFile(), 0xCAFEL);
+
+    final var newerSnapshot =
+        (FileBasedSnapshot) takeTransientSnapshot(2, snapshotStore).persist().join();
+
+    // when
+    snapshotStore.close();
+    snapshotStore = createStore(snapshotsDir, pendingSnapshotsDir);
+
+    // then
+    assertThat(snapshotStore.getLatestSnapshot()).hasValue(newerSnapshot);
```

Comment:
For completeness, I would also suggest to verify that the valid snapshot is not deleted.
```
    assertThat(newerSnapshot.getDirectory()).exists();
    assertThat(newerSnapshot.getChecksumFile()).exists();
```